Bookish.Web/Models/BarcodeViewModel.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Drawing.Imaging;
+using System.IO;
+using BarcodeLib;
+
+namespace Bookish.Web.Models
+{
+    public class BarcodeViewModel
+    {
+        public int TotalCopies { get; }
+        public string EncodedBarcode { get; }
+
+        public BarcodeViewModel(string isbn, int totalCopies)
+        {
+            TotalCopies = totalCopies;
+
+            var b = new Barcode();
+            var img = b.Encode(TYPE.CODE11, isbn);
+
+            using var stream = new MemoryStream();
+            img.Save(stream, ImageFormat.Png);
+            var imageBytes = stream.ToArray();
+            EncodedBarcode = Convert.ToBase64String(imageBytes);
```

Comment:
I think I would put the logic above into a service, e.g. a BarcodeService. Then all that needs to be passed to be passed to the view model is the base 64 string plus any info you want to display about the new book, e.g. title 🙂

Reply:
In which solution/directory should I create this file?

Reply:
I would expect your DataAccess project to have a Services folder. There I would put this new service and your existing service 🙂